### PR TITLE
V2

### PR DESCRIPTION
**Pull request by leonardaustin**
New devops code test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,36 @@
...
+2. The application is failing to connect to api.ravelin.com. e.g. `curl api.ravelin.com` but it seems to be working from the developers local code and machine. Please ssh into the box and has a look and fix.
+
+3. Please add the user `ravelin` as a root user, without a password but with the below public key as authentication.
+```ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDywrXuIj6+WlhM4Us9te1YLQAlsp3sPKtiRCt2DJf27zQ2Ig5YWLUZT+QMzWrcPK8ZjT/GspLgpwM
+GbB7YNKxosftwJ1/wJqS7wnCzfRVmUWUSLTBU/0if3oAZwUZYu9hZU6KleZwXZ6nfErdJQtYgdOdXYagcPGz6LgCTbNj1Bd9wQmIW4Gbjd7puHjxNk+
+BR+yigxJO+1rpsj1OMT0ksA7c181fR8YSVVcrEfpXtiV8I+PtlkLTuE2kx9hwgbGZilEsXNgvpsRg1yxb5W2cGDvpgDLKRd9o+28oNWl1H3+NeWRBYr
+qiqpdUSmuLpUFW9ESXp69DhyEK8uUiL8Rxv ravelin@ravelin-code-test
+```
+
+4. This particular server seems to have higer CPU utilisation and if effecting performance. It can't be the application as it is running ok on other servers. Please fix.
```
Comment (sjwhitworth) [inline README.md]: is affecting

Comment (philpearl) [inline README.md]: higer => higher

File: README.md [hunk 2]
```diff
@@ -1,31 +1,36 @@
...
-{"foo": "barr"}
+## Intructions
+- Work through the questions, in order.
+- Please do not spend more than 40 mins on any one problem.
+- Spend a maximum of 3 hours on the test.
+- Once you have completed the test please submit a readme. Ensure to include the IP address of the server and a readme of the things you tried for each question, even if you were unable to solve the problem.
+- Do not change the log destination of rsyslog or nginx
+
+## Setup
+* Provider your GCP email address to Ravelin
```
Comment (philpearl) [inline README.md]: Provider => Provide

File: README.md [hunk 3]
```diff
@@ -1,31 +1,36 @@
...
+* Check you have access to the image
+* `gcloud compute images list --project ravelin-code-test`
+* Create a new image
```
Comment (philpearl) [inline README.md]: a new image => a new instance

File: README.md [hunk 4]
```diff
@@ -1,31 +1,36 @@
...
+## Intructions
+- Work through the questions, in order.
+- Please do not spend more than 40 mins on any one problem.
+- Spend a maximum of 3 hours on the test.
+- Once you have completed the test please submit a readme. Ensure to include the IP address of the server and a readme of the things you tried for each question, even if you were unable to solve the problem.
+- Do not change the log destination of rsyslog or nginx
+
+## Setup
+* Provider your GCP email address to Ravelin
+* Check you have access to the image
```
Comment (philpearl) [inline README.md]: Check you have access to the image => Run the following command to check you have access to the machine image we've provided for the test

File: README.md [hunk 5]
```diff
@@ -1,31 +1,36 @@
...
-![alt text](https://raw.githubusercontent.com/unravelin/code-test-devops/master/network-diagram.png)
-
-## Bonus Points
-Please complete the above test manually first - in case you run out of time. If you have experience with the below technologies, for bonus points, please feel to use them and send us the code.
-* Configuration management (e.g. puppet, chef, ansible)
-* Terraform
-* Bastion (a.k.a Jump Box)
-* NAT
+## Problems
+1. Nginx has been configured by a developer; however, it seems to be broken. Please ssh into the box and fix the issue. It is suppose to be listerning to https://XX.XX.XX.XX
```
Comment (philpearl) [inline README.md]: listerning => listening

Comment (philpearl) [reply README.md]: Not sure what you mean by "https://XX.XX.XX.XX". Better to say "It should be serving https from the machine's IP address"

File: README.md [hunk 6]
```diff
@@ -1,31 +1,36 @@
...
+
+2. The application is failing to connect to api.ravelin.com. e.g. `curl api.ravelin.com` but it seems to be working from the developers local code and machine. Please ssh into the box and has a look and fix.
```
Comment (philpearl) [inline README.md]: "box and has a look" => "box and have a look"

File: README.md [hunk 7]
```diff
@@ -1,31 +1,36 @@
...
+
+3. Please add the user `ravelin` as a root user, without a password but with the below public key as authentication.
```
Comment (philpearl) [inline README.md]: "as a root user" ?? Do you mean "as a user with sudo privileges"?

File: README.md (hunk 6 above)
Comment (jonomacd) [inline README.md]: Any reason you are using our actual api hostname? I'm sure it is fine but if people are poking around curling at things I'd rather it not be our actual api.

Comment (jonomacd) [reply README.md]: This comment is showing up as outdate. It is not!

**Pull request merged**